src/virtualenv/run/plugin/base.py

```diff
@@ -15,7 +11,7 @@ class PluginLoader(object):
 
     @classmethod
     def entry_points_for(cls, key):
-        return OrderedDict((e.name, e.load()) for e in cls.entry_points().get(key, {}))
+        return OrderedDict((e.name, e.load()) for e in cls.entry_points().select(group=key))
```

Comment:
There are two performance considerations here:

1. The compatibility shim that provides backward compatibility for the dict interface (SelectableGroups) means that the `select` operation will flatten a dict before querying the groups.
2. The `select` operation needs to traverse every entry point, so is O(n) on the number of entry points.

How can we test if there's any real-world performance degradation as a result of this change?

Reply:
I compared the performance of 20.4.7 against this branch and the performance is faster on Python 3.9:

```
draft $ virtualenv foo
created virtual environment CPython3.9.5.final.0-64 in 365ms
  creator CPython3Posix(dest=/Users/jaraco/draft/foo, clear=False, no_vcs_ignore=False, global=False)
  seeder FromAppData(download=False, pip=bundle, setuptools=bundle, wheel=bundle, via=copy, app_data_dir=/Users/jaraco/Library/Application Support/virtualenv)
    added seed packages: pip==21.0.1, setuptools==57.0.0, wheel==0.36.2
  activators BashActivator,CShellActivator,FishActivator,PowerShellActivator,PythonActivator,XonshActivator
draft $ ~/p/pypa/virtualenv/.tox/python/bin/virtualenv bar
created virtual environment CPython3.9.5.final.0-64 in 323ms
  creator CPython3Posix(dest=/Users/jaraco/draft/bar, clear=False, no_vcs_ignore=False, global=False)
  seeder FromAppData(download=False, pip=bundle, setuptools=bundle, wheel=bundle, via=copy, app_data_dir=/Users/jaraco/Library/Application Support/virtualenv)
    added seed packages: pip==21.0.1, setuptools==57.0.0, wheel==0.36.2
  activators BashActivator,CShellActivator,FishActivator,PowerShellActivator,PythonActivator,XonshActivator
```

So it's unlikely a concern.